### PR TITLE
Enable lints in costmap_generator

### PR DESCRIPTION
**Pull request by nnmm**
The usual: Many header guards and includes needed fixing. clang-tidy found nothing in this package.

### CODE REVIEW

File: planning/scenario_planning/parking/costmap_generator/include/costmap_generator/costmap_generator.hpp [hunk 1]
```diff
@@ -42,11 +42,12 @@
  *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (nnmm) [inline planning/scenario_planning/parking/costmap_generator/include/costmap_generator/costmap_generator.hpp:42]: I found it a bit strange that there are two copyright headers, but they seem to be there in master already. Is that correct?

Comment (mitsudome-r) [reply planning/scenario_planning/parking/costmap_generator/include/costmap_generator/costmap_generator.hpp:42]: I think the very original code was copied from Autoware.AI, which was first made by Nagoya University, and was modified for this project. That's why we had to leave the old license for copied codes.

**Comment by simon-t4** [comment]
@nnmm Again some conflicts to resolve. Otherwise looks fine.

**Comment by nnmm** [comment]
@esteve The merge conflicts have been resolved, this should be fine to merge.

**Review by esteve — APPROVED** [review]

**Pull request merged**